src/applications/MachineShopSimulator.java

```diff
@@ -42,40 +42,55 @@ static boolean moveToNextMachine(Job theJob, SimulationResults simulationResults
         }
     }
 
+    private static int getMachine(Job theJob) {
+        return ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
+    }

```

Comment:
It seems like the message chain is still here. If something will get changed in this chain relationship, then this method will have to be changed. It also isn't easy to read and follow through.
We approached this problem in a different way, we found the end point method "getMachine()" and moved it to the beginning of the chain which in this case is "Job" class.